src/GitHub.VisualStudio/Menus/CopyLink.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.VisualStudio.Shell;
+using System;
+using System.ComponentModel.Composition;
+using System.Windows;
+using GitHub.Extensions;
+using GitHub.Services;
+
+namespace GitHub.VisualStudio.Menus
+{
+    [Export(typeof(IDynamicMenuHandler))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class CopyLink: LinkMenuBase, IDynamicMenuHandler
+    {
+        [ImportingConstructor]
+        public CopyLink([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+        }
+
+        public Guid Guid { get { return GuidList.guidContextMenuSet; } }
+        public int CmdId { get { return PkgCmdIDList.copyLinkCommand; } }
+
+        public void Activate()
+        {
+            if (!IsGitHubRepo())
+                return;
+
+            var link = GenerateLink();
+            if (link == null)
+                return;
+            Clipboard.SetText(link.AbsoluteUri);
```

Comment:
The `System.Windows` Clipboard behaves differently from the `System.Windows.Forms` one, but yeah, it can still throw completely random exceptions whenever it feels like it. I'll follow your suggestion and print out a message if it fails.